src/containers/pages/FocusInvestigation/active/index.tsx

```diff
@@ -124,8 +145,35 @@ class ActiveFocusInvestigation extends React.Component<
       .then((result: Plan[]) => fetchPlansActionCreator(result))
       .catch(err => displayError(err));
   }
-  public handleSubmit(event: React.FormEvent<HTMLFormElement>) {
-    event.preventDefault();
+
+  public debouncedSearch(event: React.ChangeEvent<HTMLInputElement>) {
+    this.setState({
+      search: event.target.value,
+    });
+
+    if (event.target.value) {
+      this.setState({
+        searchedCaseTriggeredPlans: makePlansArraySelector()(store.getState(), {
+          interventionType: InterventionType.FI,
+          parentJurisdictionId: this.props.jurisdictionParentId,
+          reason: CASE_TRIGGERED,
+          statusList: [PlanStatus.ACTIVE, PlanStatus.COMPLETE],
+          title: event.target.value,
+        }),
+        searchedRoutinePlans: makePlansArraySelector()(store.getState(), {
+          interventionType: InterventionType.FI,
+          parentJurisdictionId: this.props.jurisdictionParentId,
+          reason: ROUTINE,
+          statusList: [PlanStatus.ACTIVE, PlanStatus.COMPLETE],
+          title: event.target.value,
+        }),
+      });
+    }
+  }
```

Comment:
I think that this should probably be done in `mapStateToProps` - no?  I think that is the standard approach.

Reply:
@moshthepitt Ok...but how to we get the input value from `mapStateToProps`?

Reply:
I think that state is accessible from mapStateToProps.  Right?  If so then if the search value is in state then it can work.

Reply:
> I think that state is accessible from mapStateToProps. Right? If so then if the search value is in state then it can work.

Ok..Let me try

Reply:
@moshthepitt I think not unless we move the search value to redux store? @p-netm What to you think?

Reply:
yeah I don't hink we can access state from mapStateToProps. but we can still use the selectors inline inside the component, I think.

Reply:
I have been thinking about this, and i think if we have to supply all the prop options to the inlined selectors like done in the PR, I would go with filtering through the plans within the component.

Reply:
Basically I feel extremely uncomfortable doing the filtering within the component.  I am yet to see anything convincing about why we would want that.

Reply:
@kelvin-muchiri @moshthepitt  hmm,.. then seems as intended, the searchText will also have to be stored in the redux store, the debounced search functions implementation should only dispatch an action to modify this value. My only worry with this approach is that it feels to me like abusing the plans ducks module.

Reply:
I also think we need a final decision on this, like now. or we can create an issue.

Reply:
> abusing the plans ducks module

@p-netm how so?

Reply:
@moshthepitt taking the plans ducks for instance. it stores plans data fetched from some data source. In this regard it can be re-used anywhere where such data is in needed without assumptions as to how the component maintains its own UI state(the search string would be one this). If anything its almost nitpickish, but wanted to bring it up nevertheless.

Reply:
It is not yet clear to me whether we 100% have to store the search text in Redux.  

Additionally, keep in mind that we are going to have to support search in a lot of our components.  We should come up with something that is clean and re-usable.  In this very PR we are doing the same search functionality in three different components, for instance. 

cc: @cKellyDesign @kelvin-muchiri @kahummer

Reply:
> It is not yet clear to me whether we 100% have to store the search text in Redux.

Interesting, could we simply push it to a URL query param, which the containers would reference via the router props?

This could allow us to grab the string and pass it to our existing selectors as another filter parameter all in `mapStateToProps` (which I agree is how/where this filtering should be happening).

@moshthepitt @kelvin-muchiri @p-netm

Reply:
@moshthepitt If I understand you correctly then we are not going to store the searchString in the store, then that leaves us with atleast these 2 options:

- filter the plans inline in the component using the canonical `Array.filter`.
  - which is less than ideal
- filter the plans inline using the selectors
  - Introduce another gateway for data coming in from the store, which is even lesser as far as the ideal spectrum goes.

This is because we only have access to the searchString inside the component. AFAIK, we can only use props in mapStateToProps not the component's state

Reply:
@p-netm @kelvin-muchiri I think @cKellyDesign's input [here](https://github.com/onaio/reveal-frontend/pull/802#discussion_r411467588) is brilliant. It seems if the search terms is part of the URL we can get at least two benefits that I care a lot about:

1. We can access the URL param in mapStateToProps
2. We can now have a URL that will lead you to the filtered/searched page i.e. user's can share this URL to render already filtered pages.

Unless there is an objection, I vote that we go with that.

Thanks @cKellyDesign

Reply:
> Interesting, could we simply push it to a URL query param, which the containers would reference via the router props?

I like this option

Reply:
>  It seems if the search terms is part of the URL 

@kelvin-muchiri @p-netm my suggestion on how to best do this would be to have the search form submit a GET request to the URL in which it exists.  Doing this keeps us honest (we are using forms in the exact way they were intended) and makes it trivial to add more filter fields to the same form that work the same way.

Reply:
This is actually very similar to how we did it in Kaznet (and I bet elsewhere): https://github.com/onaio/kaznet-frontend/blob/master/src/containers/tasks/TasksList.js#L45

Reply:
@moshthepitt @cKellyDesign @p-netm I like the idea. But I think we cannot avoid using the store to store the search string. Cause we need to use the update string in `mapStateToProps`

Reply:
Updated string?

Reply:
@kelvin-muchiri any updates on the search string should be reflected in an updated url

Reply:
@cKellyDesign @p-netm Just to be on the same page. Are we going to store the searched value in redux store?

Reply:
No I don't think that will be necessary

Reply:
Ok...but `mapStateToProps` will not fire if the state hasn't changed so if we move the selectors to `mapStateToProps` and update the URL, will it really work?

Reply:
Isn't one of the stores is the state of the router, which should update upon `history.push()`?

Reply:
@cKellyDesign Yea its working

Reply:
@kelvin-muchiri How are you implementing this?  Just checking, because as I mentioned above, I think the heavy lifting should be done in the form, both because this is a natural use of HTML forms, and it makes it trivial to re-use the code.

Reply:
I am changing the search to happen on submit.

Reply:
I am retroducing the submit button

Reply:
@moshthepitt @cKellyDesign @p-netm I also realized that for most views (if not all) absence of data renders a loader rather. That means even if a call to the API was a success but  empty response was returned, the loader will still show. Which I think from a user experience point of view its confusing

Reply:
Once a call to the API is successful we should stop the loader and display the data if not display no data message

Reply:
@kelvin-muchiri 100% agreed.  That is also the subject of this discussion issue https://github.com/onaio/reveal-frontend/issues/806 - please contribute!

Reply:
> That means even if a call to the API was a success but an empty response was returned, the loader will still show.

We should very much have a standard way of knowing when a request is finished and that the UI should update accordingly. (#806)

In this case, (if we know our request has completed, and) if we know that our selector returns no plans, then we can implicitly know that there are not going to be any plans to show (at this time), so we can at that point remove the spinner, and then indicate that there are no plans matching the search parameters.

If at this point, by some asynchronous function a plan matching the filter criteria is added to the store, the container props will update, which will trigger component to re-render/return with the fresh plan showing in the table (replacing our message of sincerest apologies that no plans were found 😉).